src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  closeModal: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, closeModal }) => {
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    (async function loadUser() {
+      const userData = await getUser(userId);
+
+      setUser(userData);
+    }());
+  }, []);
```

Comment:
Move function declaration from useEffect. Just call it here. use useCallback for this async function.

Reply:
> Move function declaration from useEffect. Just call it here. use useCallback for this async function.

thanks for check, all done